src/main/java/com/adobe/target/edge/client/ondevice/collator/PageParamsCollator.java

```diff
@@ -95,26 +96,36 @@ private String extractTopLevel(String host) {
     if (host == null) {
       return "";
     }
-    int idx = host.lastIndexOf('.');
-    if (idx >= 0 && host.length() > 1) {
-      return host.substring(idx + 1);
+    String lowerCaseHost = host.toLowerCase();
+    InternetDomainName publicSuffix = InternetDomainName.from(host).publicSuffix();
+    if (publicSuffix == null) {
+      return "";
+    }
+    String lowerCaseResult = publicSuffix.toString();
+    if (lowerCaseHost.equals(host)) {
```

Comment:
this upper/lowercase code is here because `InternetDomainName.from(host)` normalizes input by changing any caps to lowercase. If preserving uppercase is necessary downstream, these checks and conversions are required